*: remove/move some dependency on `cloud`

ZhouXing19 opened the pull request:
In #86334, we need to import `descs` to the `cloud` pkg, and it caused a 
dependency loop. This PR is to break the existing `descs`'s dependency 
on `cloud`.
The 1st commit uses inject to avoid importing `cloud` to `jobspb`.
The 2nd commit removes unused methods in `kvserver` and `batcheval`
that import `cloud`.

Release justification: part of a bug fix

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87214)
<!-- Reviewable:end -->

ZhouXing19 committed:
jobspb,changefeedccl: inject jobspb.ChangefeedDetailsMarshaler

jobspb depending on cloud meant that it transitively depended on quite a
bit. serverutils depends on jobspb via serverpb. This was a mess, though
the current injection is not great either. We now let `changefeedccl` inject
`jobspb.ChangefeedDetailsMarshaler` without passing the dependency on
`cloud.SanitizeExternalStorageURI`.

Release justification: part of a bug fix
Release note: none

ZhouXing19 committed:
kvserver: remove unused `cloud` related functions

In the past we had ExportRequest/ImportRequest using these
`GetExternalStorage` methods but that logic has been long gone.
We're removing it to break the depedency of the `kvserver` and `batcheval` on
the `cloud` package.

Release justification: part of the bug fix
Release note: none
